### PR TITLE
Fixed label hitting 'x' for detail in map and padding for drawer to make it easier to open

### PR DESCRIPTION
**Pull request by eashan-mathur**
Screen Shot 2020-09-26 at 3 50 19 PM
Screen Shot 2020-09-26 at 3 50 44 PM

### CODE REVIEW

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift [hunk 1]
```diff
@@ -251,9 +255,25 @@ class OverviewCardView: CardView {
...
+//        let geoCoder = CLGeocoder()
+//        geoCoder.geocodeAddressString(myAddress!) { (placemarks, error) in
+//            guard let placemarks = placemarks?.first else { return }
+//            let location = placemarks.location?.coordinate ?? CLLocationCoordinate2D()
+//            guard let url = URL(string:"http://maps.apple.com/?daddr=\(location.latitude),\(location.longitude)") else { return }
+//            UIApplication.shared.open(url)
+//        }
+        print("tapped")
+    }
+
```
Comment (KevWho) [inline berkeley-mobile/Common/DetailView/OverviewCardView.swift:276]: I know these changes aren't doing anything right now, but could you remove them since they will be completed in a separate PR?

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift [hunk 2]
```diff
@@ -251,9 +255,25 @@ class OverviewCardView: CardView {
...
+    @objc func openAddressInMap(_ sender: UITapGestureRecognizer? = nil) {
+//        let myAddress = addressLabel.text
+//        let geoCoder = CLGeocoder()
+//        geoCoder.geocodeAddressString(myAddress!) { (placemarks, error) in
+//            guard let placemarks = placemarks?.first else { return }
+//            let location = placemarks.location?.coordinate ?? CLLocationCoordinate2D()
+//            guard let url = URL(string:"http://maps.apple.com/?daddr=\(location.latitude),\(location.longitude)") else { return }
+//            UIApplication.shared.open(url)
+//        }
+        print("tapped")
```
Comment (huangshawn1) [inline berkeley-mobile/Common/DetailView/OverviewCardView.swift:274]: we don't want to merge this unfinished/commented stuff. you can move these changes to another branch if you want to finish the address tapping in future

File: berkeley-mobile/Map/MapMarkerDetailView.swift [hunk 3]
```diff
@@ -157,6 +157,7 @@ class MapMarkerDetailView: UIView {
         verticalStack.addArrangedSubview(nameLabel)
         if (notesLabel.text?.count ?? 0) > 0 {
             verticalStack.addArrangedSubview(notesLabel)
+            nameLabel.rightAnchor.constraint(equalTo: closeButton.rightAnchor, constant: -10).isActive = true
```
Comment (KevWho) [inline berkeley-mobile/Map/MapMarkerDetailView.swift:160]: I guess that's what happens if you add a constraint to something in a stackview, haha. Maybe we could move it out of the stackview and have the stackview under it, that way we are free to constrain it however we want.

Screen Shot 2020-09-26 at 5 10 11 PM

**Review by huangshawn1 — COMMENTED** [review]
don't want to merge unfinished address stuff

File: berkeley-mobile/Map/MapMarkerDetailView.swift (hunk 3 above)
Comment (KevWho) [inline berkeley-mobile/Map/MapMarkerDetailView.swift:160]: I think you should add this line to the `init` function above, where the other UI elements are laid out. Also, you probably want to constrain `nameLabel.rightAnchor` to `closeButton.leftAnchor` (instead of `rightAnchor`). As is, it still overlaps a bit:

Screen Shot 2020-09-26 at 5 11 49 PM

**Review by KevWho — CHANGES_REQUESTED** [review]

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift (hunk 1 above)
Comment (KevWho) [reply berkeley-mobile/Common/DetailView/OverviewCardView.swift:276]: I guess Shawn beat me to it :)

File: berkeley-mobile/Map/MapMarkerDetailView.swift [hunk 4]
```diff
@@ -154,7 +159,8 @@ class MapMarkerDetailView: UIView {
             detailStack.addArrangedSubview(UIView())
         }
         
-        verticalStack.addArrangedSubview(nameLabel)
+//        verticalStack.addArrangedSubview(nameLabel)
+
```
Comment (KevWho) [inline berkeley-mobile/Map/MapMarkerDetailView.swift:163]: You can just delete this line instead of commenting it out 👍

File: berkeley-mobile/Map/MapMarkerDetailView.swift [hunk 5]
```diff
@@ -94,26 +94,31 @@ class MapMarkerDetailView: UIView {
...
-        verticalStack.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor).isActive = true
-        verticalStack.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor).isActive = true
-        
         nameLabel = UILabel()
+        contentView.addSubview(nameLabel)
         nameLabel.numberOfLines = 0
         nameLabel.font = Font.bold(20)
         nameLabel.textColor = Color.primaryText
+        nameLabel.translatesAutoresizingMaskIntoConstraints = false
+        nameLabel.topAnchor.constraint(equalTo: contentView.layoutMarginsGuide.topAnchor, constant: -offset).isActive = true
```
Comment (KevWho) [inline berkeley-mobile/Map/MapMarkerDetailView.swift:103]: Why the `constant: -offset` instead of having it flush with the layout margins?

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift [hunk 6]
```diff
@@ -25,6 +27,7 @@ enum OverviewElements {
 class OverviewCardView: CardView {
     var item: SearchItem!
     var addressView: UIView?
+//    let tap = UITapGestureRecognizer(target: self, action: #selector(self.openAddressInMap(_:)))
```
Comment (KevWho) [inline berkeley-mobile/Common/DetailView/OverviewCardView.swift:30]: You can also delete this comment 👍

File: berkeley-mobile/Drawer/MainDrawerViewController.swift [hunk 7]
```diff
@@ -14,7 +14,7 @@ class MainDrawerViewController: DrawerViewController {
     var bottomOffset: CGFloat? {
         didSet {
             tabBarViewController.view.frame = self.view.bounds.inset(by: UIEdgeInsets(top: 0, left: 0, bottom: bottomOffset! + 20, right: 0))
-            tabBarViewController.view.frame.origin.y = barView.frame.maxY + 16
+            tabBarViewController.view.frame.origin.y = barView.frame.maxY + 24
```
Comment (KevWho) [inline berkeley-mobile/Drawer/MainDrawerViewController.swift:17]: This seems to remove the small bit of bottom padding originally below the table views. Could you adjust the frame of `tabBarViewController` (in the line above this one) to re-add the padding?
Screen Shot 2020-10-04 at 4 41 33 PM

**Review by KevWho — CHANGES_REQUESTED** [review]
Nice work! Just requested a couple very small changes.

File: berkeley-mobile/Map/MapMarkerDetailView.swift (hunk 5 above)
Comment (Stereopoly) [reply berkeley-mobile/Map/MapMarkerDetailView.swift:103]: Offset to match vertially align nameLabel with the "x"

**Review by KevWho — APPROVED** [review]
Looks good, thanks!
